internal/environments/environments.go

```diff
@@ -107,6 +107,12 @@ func (l *Landscape) AddSystem(system *SAPSystem) {
 	l.SAPSystems[system.Name] = system
 }
 
+func (l *Landscape) AddSystems(systems []*SAPSystem) {
```

Comment:
This new method is not used anywhere. I think you created this to replace the loop used above maybe, at line 62 of `agent/discovery/sapsystem.go`.
You could whether remove this function or use it above replacing the loop (you might need to change the incoming variable type)